display_list/geometry/dl_path.cc

```diff
@@ -0,0 +1,149 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/display_list/geometry/dl_path.h"
+
+#include "flutter/display_list/geometry/dl_geometry_types.h"
+#include "flutter/impeller/geometry/path_builder.h"
+
+namespace flutter {
+
+const SkPath& DlPath::GetSkPath() const {
+  return sk_path_;
+}
+
+impeller::Path DlPath::GetPath() const {
+  if (path_.IsEmpty() && !sk_path_.isEmpty()) {
+    path_ = ConvertToImpellerPath(sk_path_);
+  }
+
+  return path_;
+}
+
+bool DlPath::IsInverseFillType() const {
+  return sk_path_.isInverseFillType();
+}
+
+bool DlPath::IsRect(DlRect* rect, bool* is_closed) const {
```

Comment:
I think we only actually care about closed rects right? This could handle checking is_closed internally.

Reply:
For fill it doesn't matter if it is closed, for stroke then you care.